### PR TITLE
Merge account_baking_payment_export module into account_payment_order. #68

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
#68

File: account_payment_order/README.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+
+
+Payment order
+============================
```
Comment (pedrobaeza) [inline account_payment_order/README.rst:7]: The lenght of the line should be the same as the title length. Please copy the same line above the title for highlighting the title more than the sections.

File: account_payment_order/README.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+This module brings back the payment.order from v8.
+
+This module also provide an infrastructure to export payment orders.
+It includes some bug fixes and obvious enhancements to payment orders that will hopefully land in offical addons one
```
Comment (pedrobaeza) [inline account_payment_order/README.rst:12]: You can remove this text, as it refers to v8 version.

**Comment by pedrobaeza** [comment]
You must preserve commit history. See this similar guide for knowing how to do it: https://github.com/odoomrp/odoomrp-wip/wiki/Migration-to-OCA

File: account_payment_order/README.rst [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+day.
+This technical module provides the base infrastructure to export payment orders
```
Comment (pedrobaeza) [inline account_payment_order/README.rst:14]: This text was specific for the account_banking_payment_export. I think we can remove it.

File: account_payment_order/README.rst [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+* base_iban
+
+This modules is part of the OCA/account-payment suite.
+
+
+Usage
+=====
+
+This module provides a menu to configure payment order types : Accounting > Configuration > Miscellaneous > Payment Export Types
```
Comment (pedrobaeza) [inline account_payment_order/README.rst:42]: This can be in the Configuration section.

File: account_payment_order/README.rst [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+This modules is part of the OCA/account-payment suite.
+
+
+Usage
+=====
+
+This module provides a menu to configure payment order types : Accounting > Configuration > Miscellaneous > Payment Export Types 
+
+For further information, please visit:
```
Comment (pedrobaeza) [inline account_payment_order/README.rst:44]: Remove these lines that are obsolete

File: account_payment_order/README.rst [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+   :target: https://runbot.odoo-community.org/runbot/125/9.0
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/l10n-switzerland/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/l10n-switzerland/issues/new?body=module:%20l10n_ch_states%0Aversion:%208.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (pedrobaeza) [inline account_payment_order/README.rst:59]: This should change accordingly new repo

File: account_payment_order/README.rst [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+This module provides a menu to configure payment order types : Accounting > Configuration > Miscellaneous > Payment Export Types 
+
+For further information, please visit:
+
+ * https://www.odoo.com/forum/help-1
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/125/9.0
```
Comment (pedrobaeza) [inline account_payment_order/README.rst:50]: 125 is the account-payment or l10n-switzerland repo?

File: account_payment_order/__init__.py [hunk 8]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 brain-tec AG (http://www.braintec-group.com)
```
Comment (pedrobaeza) [inline account_payment_order/__init__.py:2]: You should put here all the companies or not put anyone (best option, as they are too much). But in main files you should include all of them with their respective dates.

File: account_payment_order/__openerp__.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 brain-tec AG (http://www.braintec-group.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Payment Order',
+    'summary': 'Port of account_payment from v8',
+    'version': '9.0.1.0.1',
+    'author': "Odoo Community Association (OCA), brain-tec AG",
```
Comment (pedrobaeza) [inline account_payment_order/__openerp__.py:9]: You should include here all the involved companies, Odoo included.

File: account_payment_order/__openerp__.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    'category': 'Accounting & Finance',
+    'website': 'https://github.com/OCA/bank-payment',
+    'license': 'AGPL-3',
+    'depends': [
+        'base',
+        'account',
+        'document',
+        'account_full_reconcile',
+        'base_iban',  # for manual_bank_transfer
+        'account_full_reconcile',
```
Comment (pedrobaeza) [inline account_payment_order/__openerp__.py]: For what is this module needed?

File: account_payment_order/i18n/account_payment_order.pot [hunk 11]
```diff
@@ -0,0 +1,276 @@
+# Translation of Odoo Server.
```
Comment (pedrobaeza) [inline account_payment_order/i18n/account_payment_order.pot:1]: This file is not needed anymore

File: account_payment_order/models/payment_line.py [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+                }
+                pl.amount = pl.with_context(ctx).currency_id.\
+                    compute(pl.amount_currency, pl.company_currency)
+
+    name = fields.Char('Your Reference', required=True, default=lambda self:
+                       self.env['ir.sequence'].next_by_code('payment.line'),
+                       copy=False
+                       )
+
+    communication = fields.Char('Communication', required=True,
```
Comment (pedrobaeza) [inline account_payment_order/models/payment_line.py:62]: You can save a lot of lines indenting this way:

```
communication = fields.Char(
    'Communication', required=True,
    help="...
```

File: account_payment_order/models/payment_mode.py [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+            model_data.create(
+                cr, uid,
+                {'module': 'account_payment_order',
+                 'model': 'payment.mode.type',
+                 'name': 'manual_bank_tranfer',
+                 'res_id': res,
+                 'noupdate': False})
+        return res
+
+    def _auto_init(self, cr, context=None):
```
Comment (pedrobaeza) [inline account_payment_order/models/payment_mode.py]: This is not needed anymore, as it's no more an inherited model.

File: account_payment_order/models/payment_mode_type.py [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+        domain=[('osv_memory', '=', True)])
+    payment_order_type = fields.Selection(
+        [('payment', 'Payment'),
+         ('debit', 'Debit')],
+        string='Order type', required=True, default='payment',
+        help="This field determines if this type applies to customers "
+             "(Debit) or suppliers (Payment)")
+    active = fields.Boolean(string='Active', default=True)
+
+    def _auto_init(self, cr, context=None):
```
Comment (pedrobaeza) [inline account_payment_order/models/payment_mode_type.py]: This code can be removed also as it was a migration trick from v7 to v8

File: account_payment_order/models/payment_order.py [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+
+
+class PaymentOrder(models.Model):
+    _name = 'payment.order'
+    _description = 'Payment Order'
+    _rec_name = 'reference'
+    _order = 'id desc'
+
+    @api.depends('line_ids', 'line_ids.amount')
+    @api.one
```
Comment (pedrobaeza) [inline account_payment_order/models/payment_order.py:16]: @api.one is deprecated. Use @api.multi.

File: account_payment_order/models/payment_order.py [hunk 16]
```diff
@@ -0,0 +1,172 @@
...
+        'Payment order type', required=True, default='payment',
+        readonly=True, states={'draft': [('readonly', False)]})
+
+    mode_type = fields.Many2one('payment.mode.type', related='mode.type',
+                                string='Payment Type')
+
+    @api.one
+    def set_to_draft(self):
+        self.write({'state': 'draft'})
+#         self.create_workflow()
```
Comment (pedrobaeza) [inline account_payment_order/models/payment_order.py:85]: I see that you have removed workflow. Wise decision! It's a nonsense here. But please remove all commented lines from the final result. When you have migrated all the commit history, we were been able to see the removal in the diff

File: account_payment_order/security/ir.model.access.csv [hunk 17]
```diff
@@ -0,0 +1,13 @@
+"id","name","model_id:id","group_id:id","perm_read","perm_write","perm_create","perm_unlink"
+,,,,,,,
```
Comment (pedrobaeza) [inline account_payment_order/security/ir.model.access.csv:2]: Why these empty lines?

File: account_payment_order/wizard/payment_order_create.py [hunk 18]
```diff
@@ -0,0 +1,234 @@
...
+    bank account type,
+    - Big number of partner and bank account.
+
+    If a type is given, unsuitable account Entry lines are ignored.
+    """
+
+    _name = 'payment.order.create'
+    _description = 'payment.order.create'
+
+    duedate = fields.Date('Due Date', required=True,
```
Comment (pedrobaeza) [inline account_payment_order/wizard/payment_order_create.py:25]: As this is a TransientModel, there's no problem in renaming some columns, so I suggest to rename this to `due_date` for being more readable

File: account_payment_order/wizard/payment_order_create.py [hunk 19]
```diff
@@ -0,0 +1,234 @@
...
+                          default=fields.Date.today)
+
+    entries = fields.Many2many('account.move.line')
+
+    populate_results = fields.Boolean(string="Populate results directly",
```
Comment (pedrobaeza) [inline account_payment_order/wizard/payment_order_create.py:30]: I'm missing here a lot of changes from the last version of account_banking_payment_export. Is this not updated?

File: account_payment_order/wizard/payment_order_create.py [hunk 20]
```diff
@@ -0,0 +1,234 @@
...
+                                      default=True)
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False,
```
Comment (pedrobaeza) [inline account_payment_order/wizard/payment_order_create.py:34]: You can do directly this change over the view, as now it's not inherited anymore

**Comment by pedrobaeza** [comment]
@BT-kaberer, thanks for proposing the PR. There are some comments inline, but you have to re-work a bit the proposal.

I'm also seeing that merging the account_direct_debit module here makes all sense, as we don't need more to keep the modularity (it was because of the original account_payment module doesn't handle direct debit orders).

**Comment by pedrobaeza** [comment]
I'm going to start again this work because this one has been based on a deprecated version on bank-payment suite. Preserve commit history is going to be insane, because only account_payment module from Odoo has 3926 commits!, but the authors are going to be credited.

**Pull request closed without merging**